src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -165,14 +165,12 @@ class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: I
 
   override def reverse = {
     val r = new ArrayDeque[A](initialSize = size)
-    var i = length
-    while(0 < i) {
-      i -= 1
-      r.appendAssumingCapacity(this(i))
-    }
+    this.foreach(r.prependAssumingCapacity)
```

Comment:
the foreach simply is the while loop in `IndexedSeqOptimized`:

```scala
override /*IterableLike*/
  def foreach[U](f: A => U): Unit = {
    var i = 0
    val len = length
    while (i < len) { f(this(i)); i += 1 }
  }
```

I think this should be fine...